zephyr/wrapper.c

```diff
@@ -230,7 +234,11 @@ void rfree(void *ptr)
 
 	/* select heap based on address range */
 	if (is_uncached(ptr)) {
+#ifdef CONFIG_IMX
+		heap_free(&sof_heap, ptr);
```

Comment:
would you ever get uncached pointers from malloc on IMX? From your description it sounds like you don't need those. Let's do this: let's get one of these PRs in - either this one or #4479 (I don't mind either, but I'd prefer mine because it's smaller, it only fixes a bug without adding more confusion). After that I'd make a new PR to extract all heap management into a separate file zephyr/alloc.c and then we'd duplicate it into zephyr/alloc_cavs.c and zephyr/alloc_imx.c or whatever. And remove all the `#ifdef`s from both. Then you can do what you need in your copy.

Reply:
> would you ever get uncached pointers from malloc on IMX?

We don't have the memory map aliasing that permits cached vs uncached pointers. I was having internal discussions more than a year ago about this, currently cache_to_uncache and uncache_to_cache macros are no-ops but I was wondering if there was a way to remove them (because they cannot be meaningfully implemented) for platforms like IMX.